chrome/content/zotero/xpcom/utilities.js

```diff
@@ -1989,7 +1989,54 @@ Zotero.Utilities = {
 	 * Provides unicode support and other additional features for regular expressions
 	 * See https://github.com/slevithan/xregexp for usage
 	 */
-	 "XRegExp": typeof XRegExp !== "undefined" ? XRegExp : null
+	"XRegExp": typeof XRegExp !== "undefined" ? XRegExp : null,
+
+	/**
+	 * Cleans an edition string ("second edition", "3e", "4."), converting
+	 * spelled-out ordinals and returning only the numeric part ("2", "3", "4").
+	 * 
+	 * If text is null or empty, or no edition can be found, returns the passed text.
+	 * If the resolved edition is "1" ("first edition", "1. Auflage"), returns null.
+	 * The Zotero.Item edition field should be left empty for first-edition items.
+	 * 
+	 * @param {string} text 
+	 * @param {boolean} allowOtherText If true, match even strings with extra content
+	 * ("3rd edition of the book", "there are 3 fish" -> "3"). More flexible, but can
+	 * cause false positives or lose information ("2nd edition with the author's signature" -> "2").
+	 * @return {string}
+	 */
+	"cleanEdition": function (text, allowOtherText) {
```

Comment:
Let's move this up next to the other `clean*` functions (and move the test up accordingly).